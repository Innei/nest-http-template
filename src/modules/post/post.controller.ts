import { Controller, Get, Query } from '@nestjs/common'

import { HTTPDecorators } from '~/common/decorator/http.decorator.js'
import { ApiName } from '~/common/decorator/openapi.decorator.js'
import { BizException } from '~/common/exceptions/business.excpetion.js'
import { ErrorCodeEnum } from '~/constants/error-code.constant.js'
import { PagerDto } from '~/shared/dto/pager.dto.js'

import { PostService } from './post.service.js'

@Controller('posts')
@ApiName
export class PostController {
  constructor(private readonly service: PostService) {}

  @HTTPDecorators.Paginator
  @Get('/')
  async gets(@Query() query: PagerDto) {
    const { page, select, size } = query
    return this.service.model.paginate(
      {},
      {
        page,
        limit: size,
        select,
        lean: true,
      },
    )
  }

  @Get('/*')
  async notFound() {
    throw new BizException(ErrorCodeEnum.PostNotFoundError)
  }
}
