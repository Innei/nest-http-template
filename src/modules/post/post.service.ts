import { Injectable } from '@nestjs/common'

import { InjectModel } from '~/transformers/model.transformer.js'

import { PostModel } from './post.model.js'

@Injectable()
export class PostService {
  constructor(
    @InjectModel(PostModel) public readonly model: MongooseModel<PostModel>,
  ) {}
}
