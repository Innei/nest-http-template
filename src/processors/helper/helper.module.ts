import { Global, Module, Provider } from '@nestjs/common'

import { HttpService } from './helper.http.service.js'

const providers: Provider<any>[] = [HttpService]

@Module({
  imports: [],
  providers,
  exports: providers,
})
@Global()
export class HelperModule {}
