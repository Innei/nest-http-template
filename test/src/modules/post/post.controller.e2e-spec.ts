import { createE2EApp } from 'test/helper/create-e2e-app'
import { clearDynamicData } from 'test/utils'

import { PostController } from '~/modules/post/post.controller'
import { PostModel } from '~/modules/post/post.model'
import { PostService } from '~/modules/post/post.service'

describe('Test PostController E2E', () => {
  const proxy = createE2EApp({
    controllers: [PostController],
    providers: [PostService],
    models: [PostModel],
    async pourData(modelMap) {
      const { model } = modelMap.get(PostModel)
      await model.create([
        {
          title: 'test1',
          text: 'test1',
          slug: 'test1',
        },
        {
          title: 'test2',
          text: 'test2',
          slug: 'test2',
        },
      ])

      return async () => {
        await model.deleteMany({})
      }
    },
  })

  test('GET /posts', async () => {
    const app = proxy.app

    const res = await app.inject({
      method: 'GET',
      url: '/posts',
    })

    expect(res.statusCode).toBe(200)
    const result = res.json()
    clearDynamicData(result.data)
    expect(result).toMatchSnapshot()
  })
})
