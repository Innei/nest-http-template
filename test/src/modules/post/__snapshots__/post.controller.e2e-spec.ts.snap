// Vitest Snapshot v1

exports[`Test PostController E2E > GET /posts 1`] = `
{
  "data": [
    {
      "slug": "test1",
      "text": "test1",
      "title": "test1",
    },
    {
      "slug": "test2",
      "text": "test2",
      "title": "test2",
    },
  ],
  "pagination": {
    "current_page": 1,
    "has_next_page": false,
    "has_prev_page": false,
    "size": 10,
    "total": 2,
    "total_page": 1,
  },
}
`;
